docs/table-migration.md

```diff
@@ -0,0 +1,89 @@
+
+## Migration Approach
+There are two primary methods for executing table migration: full data migration and in-place migration. In this doc, we will describe more about in-place migration.
+
+In-place migration preserves the existing data files while incorporating Iceberg metadata on top of them.
+This method is not only faster but also eliminates the need for data duplication, making it more appropriate for production environments.
+
+![In-Place Migration](../../../img/iceberg-in-place-migration.png)
+
+Apache Iceberg primarily supports the in-place migration approach, which includes three important actions: `Snapshot Table`, `Migrate Table`, and `Add Files`.
+
+## Snapshot Table
+The Snapshot Table action creates a new iceberg table with the same schema and partitioning as the source table, leaving the source table unchanged during and after the action.
+
+- Create a new Iceberg table with the same metadata (schema, partition spec, etc.) as the source table
+
+![Snapshot Table Step 1](../../../img/iceberg-snapshotaction-step1.png)
+
+- Commit all data files across all partitions to the new Iceberg table. The source table remains unchanged.
+
+![Snapshot Table Step 2](../../../img/iceberg-snapshotaction-step2.png)
+## Migrate Table
+The Migrate Table action also creates a new Iceberg table with the same schema and partitioning as the source table. However, during the action execution, it locks and drops the source table from the catalog.
+Consequently, Migrate Table requires all modifications working on the source table to be stopped before the action is performed.
+
+Stop all writers interacting with the source table. Readers that also supports Iceberg may continue reading.
+
+![Migrate Table Step 1](../../../img/iceberg-migrateaction-step1.png)
+
+- Create a new Iceberg table with the same identifier and metadata (schema, partition spec, etc.) as the source table. Rename the source table for a backup in case of failure and rollback.
+
+![Migrate Table Step 2](../../../img/iceberg-migrateaction-step2.png)
+
+- Commit all data files across all partitions to the new Iceberg table. Drop the source table.
+
+![Migrate Table Step 3](../../../img/iceberg-migrateaction-step3.png)
+## Add Files
+After the initial step (either Snapshot Table or Migrate Table), it is common to find some data files that have not been migrated. These files often originate from concurrent writers who continue writing to the source table during or after the migration process.
+In practice, these files can be new data files in Hive tables or new snapshots (versions) of Delta Lake tables. The Add Files action is essential for incorporating these files into the Iceberg table.
+
+## In-Place Migration Completion
+Once all data files have been migrated and there are no more concurrent writers writing to the source table, the migration process is complete.
+Readers and writers can now switch to the new Iceberg table for their operations.
```

Comment:
aren't readers and writers already switched during your process?

Reply:
Sorry for the confusion. My intention here is to mostly describe the combination of `snapshot table` and `add files`. Basically, I want to express that after snapshotting the initial snapshot and add all subsequent data files to make the new iceberg table have the same contents as the source table, users can also migrate readers and writers to the new iceberg table.

Think twice about the above process. I found that 1. the process is mostly described in the `add files` action. 2. the  migration of reader/writer after the snapshot process may be out-of-scope. Hence, I decided to delete this section.
What do you think of this change?